app/models/user.rb

```diff
@@ -0,0 +1,35 @@
+require 'data_mapper'
+require 'bcrypt'
+
+class User
+
+  include DataMapper::Resource
+
+  property :id, Serial
+  property :email, String, required: true
+  property :password_digest, Text
+
+
+  def password=(password)
+    @password = password
+    self.password_digest = BCrypt::Password.create(password)
+  end
+
+  attr_reader :password
+  attr_accessor :password_confirmation
+
+  validates_confirmation_of :password
```

Comment:
Prefer the new style validations `validates :column, confirmation: value` over `validates_confirmation_of`.